packages/amplify-category-api/src/provider-utils/awscloudformation/pipeline-with-awaiter.ts

```diff
@@ -246,11 +247,17 @@ export class PipelineWithAwaiter extends cdk.Construct {
         actions: [
           new codepipelineactions.EcsDeployAction({
             actionName: 'Deploy',
-            service: new (class extends cdk.Construct implements ecs.IBaseService {
+            service: new (class extends Construct implements ecs.IBaseService {
+              // eslint-disable-next-line class-methods-use-this
+              applyRemovalPolicy(_: cdk.RemovalPolicy): void {
+                // NO-OP
+              }
+
```

Comment:
This is forced upon us by `cdk.IResource` . I have not found an obvious candidate to delegate this call to this anonymous class members (or containing class). Therefore I'm logging follow up for this here . https://app.asana.com/0/1202389680978480/1203167795285200/f .

See https://github.com/aws/aws-cdk/blob/4525be85ff89a9606b5891d5a5a33438e57460e2/packages/%40aws-cdk/core/lib/resource.ts#L61-L72